pyflakes/test/test_imports.py

```diff
@@ -564,6 +563,15 @@ def bar():
         fu
         ''', m.UnusedImport, m.UndefinedName)
 
+    def test_shadowedInNestedScope(self):
+        self.flakes('''
+        import fu
+        def bar():
+            import fu
+            fu
+        fu
+        ''', m.ImportShadowedByImport)
```

Comment:
Just noting that in the case of the class attribute, the same can be achieved without shadowing using

```
import foo

class bar:
    foo = foo

```

But as said earlier, I am happy to allow shadowing for class attributes, and other instances if there is consensus one way or the other.